src/test/java/repository/CourseMongoRepositoryTest.java

```diff
@@ -0,0 +1,184 @@
+
+
+	private static final String DB_NAME = "schoolagenda";
+	private static final String DB_COLLECTION = "courses";
+
+	@BeforeClass
+	public static void initServer() {
+		mongoServer = new MongoServer(new MemoryBackend());
+		serverAddres = mongoServer.bind();
+	}
+
+	@AfterClass
+	public static void shutdownServer() {
+		mongoServer.shutdown();
+	}
+
+	@Before
+	public void setup() {
+		mongoClient = new MongoClient(new ServerAddress(serverAddres));
+		courseMongoRepository = new CourseMongoRepository(mongoClient, DB_NAME, DB_COLLECTION);
+		mongoClient.getDatabase(DB_NAME).drop();
+		courseCollection = mongoClient.getDatabase(DB_NAME).getCollection(DB_COLLECTION);
+	}
+
+	@After
+	public void closeServer() {
+		mongoClient.close();
+	}
+
+	@Test
+	public void testFindAllCoursesWhenCollectionIsEmptyShouldReturnEmptyList() {
+		assertThat(courseMongoRepository.findAll()).isEqualTo(Collections.emptyList());
+	}
+
+	@Test
+	public void testFindAllCoursesWhenCollectionIsNotEmptyShoudReturnStudentList() {
+		//setup
+		addTestCourseToDatabase("id", "testCourse", Collections.emptyList());
+
+		//exercise
+		List<Course> courses = courseMongoRepository.findAll();
+
+		//verify
+		assertThat(courses).containsExactly(new Course("id", "testCourse"));
+	}
+	
+	@Test
+	public void testFindCourseByIdShouldNotBeFound() {
+		//verify
+		assertThat(courseMongoRepository.findById("id")).isNull();
+	}
+	
+	@Test
+	public void testFindCourseByIdShouldBeFound() {
+		//setup
+		addTestCourseToDatabase("id", "testCourse", Collections.emptyList());
+
+		//verify
+		assertThat(courseMongoRepository.findById("id"))
+		.isEqualTo(new Course("id", "testCourse"));
+	}
+	
+	@Test
+	public void testSave() {
+		//setup
+		Course testCourse = new Course("id", "testCourse");
+
+		//exercise
+		courseMongoRepository.save(testCourse);
+
+		//verify
+		assertThat(readAllCoursesFromDatabase()).containsExactly(testCourse);
+	}
+
+	@Test
+	public void testDelete() {
+		//setup
+		Course testCourse = new Course("id", "testCourse");
+
+		//exercise
+		courseMongoRepository.delete(testCourse);
+
+		//verify
+		assertThat(readAllCoursesFromDatabase()).isEmpty();
+	}
+	
+	@Test
+	public void testGetStudentsFromCourseWhenStudentListIsEmpty() {
+		//setup
+		addTestCourseToDatabase("idCourse", "testCourse", Collections.emptyList());
+
+		//exercise
+		List<String> courseStudents = courseMongoRepository.findCourseStudents("idCourse");
+
+		//verify
+		assertThat(courseStudents).isEqualTo(Collections.emptyList());
+	}
+	
+	@Test
+	public void testGetStudentsFromCourseWhenStudentListIsNotEmpty() {
+		//setup
+		addTestCourseToDatabase("idCourse", "testCourse", Collections.singletonList("idStudent"));
+
+		//exercise
+		List<String> courseStudents = courseMongoRepository.findCourseStudents("idCourse");
+
+		//verify
+		assertThat(courseStudents).isEqualTo(Collections.singletonList("idStudent"));
+	}
+	
+	@Test
+	public void testAddStudentToCourseWhenStudentIsNotNull() {
```

Comment:
What about `testAddStudentToCourseWhenStudentIsNull`?

Reply:
Yes, I forgot to test that case. The updateCourseStudents should go in just fine and the findCourseStudents should return an emptyList()?

Reply:
Maybe throw an Exception is the best solution here. I don't know if this change will break some previous tests.